cider/exec/nextgen/operators/SourceNode.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef NEXTGEN_OPERATORS_SOURCENODE_H
+#define NEXTGEN_OPERATORS_SOURCENODE_H
+
+#include <vector>
+
+#include "exec/nextgen/operators/OpNode.h"
+
+class InputColDescriptor;
+namespace cider::exec::nextgen::operators {
+
+class SourceNode : public OpNode {
+ public:
+  SourceNode(const ExprPtrVector& col_exprs);
```

Comment:
so these `col_exprs` are all `ColumnVar`? For join, will it differentiate the left columns and right columns?

Reply:
Yes, there may be some difference in JOIN cases.